lib/modules/manager/dockerfile/extract.ts

```diff
@@ -236,6 +236,10 @@ export function getDep(
     }
   }
 
+  if (dep.depName) {
```

Comment:
why conditional?

Reply:
First it wasn't conditional, but I noticed a bunch of the getDep unittests failing, especially the error cases, and in those cases the depName isn't set either, so to preserve existing behavior and prevent issues I made it conditional.

But I have to admit I was in doubt, so I can change it if you think it's better?

Reply:
Sorry to chime in here but I don't think this makes sense. 

Conceptually, in the `dockerfile` manager, `depType` _stage_ should be set for all extracted images, except for the last one. This is unrelated to `depName`. Tests are currently failing because other managers know see a `depType` that shouldn't be there. In other words, it shouldn't be set in `getDep()` because this is not exclusively used by the `dockerfile` manager.

I'd simply iterate over all deps (as done before this PR) but exlude the first one (e.g., via `deps.slice(1)`) if it already has a `depType` set. `syntax` can only occur at the beginning of a Dockerfile, so it can only occur in the first dep, if at all.